lib/wanda/results.ex

```diff
@@ -69,6 +65,26 @@ defmodule Wanda.Results do
     |> Repo.one()
   end
 
+  @doc """
+  Marks a previously started execution as completed
+  """
+  @spec complete_execution_result!(Result.t()) :: ExecutionResult.t()
+  def complete_execution_result!(%Result{execution_id: execution_id} = result) do
+    case get_execution_result!(execution_id) do
+      %ExecutionResult{
```

Comment:
Do we need the `get_execution_result` and the case here? Isn't the update idempotent if we try to change the same field with `complete` several times? Also making a query before updating gives room for [time to check time to use bugs](https://en.wikipedia.org/wiki/Time-of-check_to_time-of-use)

Reply:
```elixir
%ExecutionResult{id: id}
|> ExecutionResult.changeset(%{
      payload: result,
      status: :completed,
      completed_at: DateTime.utc_now()
    })
|> Repo.update!()
```